src/win/thread.c

```diff
@@ -124,6 +124,33 @@ static UINT __stdcall uv__thread_start(void* arg) {
   return 0;
 }
 
+const DWORD MS_VC_EXCEPTION = 0x406D1388;
+#pragma pack(push, 8)
+  typedef struct tagTHREADNAME_INFO
+  {
+    DWORD dwType;     // Must be 0x1000.
+    LPCSTR szName;    // Pointer to name (in user addr space).
+    DWORD dwThreadID; // Thread ID (-1=caller thread).
+    DWORD dwFlags;    // Reserved for future use, must be zero.
+  } THREADNAME_INFO;
+#pragma pack(pop)
+
+void uv_thread_set_name(const char *threadName) {
+  THREADNAME_INFO info;
+  info.dwType = 0x1000;
+  info.szName = threadName;
+  info.dwThreadID = -1;
+  info.dwFlags = 0;
+
+#pragma warning(push)
+#pragma warning(disable : 6320 6322)
+  __try {
+    RaiseException(MS_VC_EXCEPTION, 0, sizeof(info) / sizeof(ULONG_PTR), (ULONG_PTR *)&info);
+  }
+  __except (EXCEPTION_EXECUTE_HANDLER) {}
+#pragma warning(pop)
+}
```

Comment:
😢 
Quoting something I saw on the interwebs WRT to this _"Yay, MSDN! Only the best samples using 20 year-old technology."_

FYI for windows >= Windows 10.1607 we now have ['SetThreadDescription'](https://msdn.microsoft.com/en-us/library/windows/desktop/mt774976(v=vs.85).aspx)

Reply:
P.S. this does not store any information on the thread, it only allows an attached debugger catch the exception and use the `THREADNAME_INFO`.
So without a debugger attached there isn't a way to implement `uv_thread_get_name` (without our own storage).